internal/goflags/flags.go

```diff
@@ -0,0 +1,168 @@
+// Package goflags allows parsing go command invocations and storing their flags in a
+// CommandFlags structure. It also provides utilities to backtrack through the process stack to
+// find and parse the flags of the first parent go command found in the process hierarchy.
+package goflags
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"github.com/datadog/orchestrion/internal/goproxy"
+	"github.com/shirou/gopsutil/v3/process"
+)
+
+// CommandFlags represents the flags provided to a go command invocation
+type CommandFlags struct {
+	Long    map[string]string
+	Short   []string
+	Unknown []string // flags we don't process but store anyway
+}
+
+var (
+	// specialFlags are long Go command flags that accept other flags as parameters
+	specialFlags = map[string]struct{}{
+		"-asmflags": {}, "-gccgoflags": {}, "-gcflags": {}, "-ldflags": {},
+	}
+	shortFlags = map[string]struct{}{
+		"-asan": {}, "-cover": {}, "-linkshared": {}, "-modcacherw": {}, "-msan": {}, "-race": {}, "-trimpath": {},
+	}
+	longFlags = map[string]struct{}{
+		"-covermode": {}, "-coverpkg": {}, "-buildmode": {}, "-buildvcs": {}, "-compiler": {}, "-mod": {},
+		"-modfile": {}, "-overlay": {}, "-pgo": {}, "-pkgdir": {}, "-tags": {},
+	}
+)
+
+// Trim removes the specified flags and their value from the long and short flags
+func (f CommandFlags) Trim(flags ...string) {
+	for _, flag := range flags {
+		delete(f.Long, flag)
+		for i, fl := range f.Short {
+			if fl == flag {
+				f.Short = append(f.Short[:i], f.Short[i+1:]...)
+				break
+			}
+		}
+	}
+}
+
+// Slice returns the command flags as a string slice
+// - long flags are returned as a string of the form '-flagName="flagVal"'
+// - short flags are returned as a string of the form '-flagName'
+// - unknow flags and values are ignored
+func (f CommandFlags) Slice() []string {
+	flags := make([]string, 0, len(f.Long)+len(f.Short))
+	for flag, val := range f.Long {
+		flags = append(flags, fmt.Sprintf("%s=\"%s\"", flag, val))
```

Comment:
I'd use `%q` to make sure nested quotes are properly handled...

```suggestion
		flags = append(flags, fmt.Sprintf("%s=%q", flag, val))
```